### PR TITLE
Add tests with Aqua.jl

### PR DESCRIPTION
**Pull request by hyrodium**

### CODE REVIEW

**Comment by kimikage** [comment]
I'm going to merge PR #278 first to pass the nightly build tests.

File: Project.toml [hunk 1]
```diff
@@ -13,17 +13,22 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 FixedPointNumbersStatisticsExt = "Statistics"
 
 [compat]
+Aqua = "0.8"
+Documenter = "0.27, 1"
```
Comment (hyrodium) [inline Project.toml:17]: We can drop the test dependency on Documenter.jl by adding documentation with Documenter.jl I will open a PR for that later.

File: test/runtests.jl [hunk 2]
```diff
@@ -1,11 +1,13 @@
-using FixedPointNumbers, Test
+using FixedPointNumbers, Test, Aqua
 
 if VERSION >= v"1.6.0-DEV.816" # JuliaLang/julia #36962 # FIXME
     @test isempty(detect_ambiguities(FixedPointNumbers))
 else
     @test isempty(detect_ambiguities(FixedPointNumbers, Base, Core))
 end
 
+Aqua.test_all(FixedPointNumbers)
```
Comment (kimikage) [inline test/runtests.jl:9]: IIRC, Aqua includes the test for ambiguity, doesn't it?

File: README.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 [![Build Status][action-img]][action-url]
 [![Build Status][pkgeval-img]][pkgeval-url]
 [![coverage][codecov-img]][codecov-url]
+[![Aqua QA](https://raw.githubusercontent.com/JuliaTesting/Aqua.jl/master/badge.svg)](https://github.com/JuliaTesting/Aqua.jl)
```
Comment (kimikage) [inline README.md:6]: Aqua.jl?

File: test/runtests.jl (hunk 2 above)
Comment (hyrodium) [reply test/runtests.jl:3]: Ohh, that's right. Updated in 40c5e55. (Removing `isempty(detect_ambiguities(FixedPointNumbers))` was my first motivation for this PR :joy:)

**Comment by kimikage** [comment]
Most of the CI errors seem to be due to Codecov's limitations of token-less upload.
The cause of the julia-invalidations error has not been identified, but is unrelated to this PR.

**Review by kimikage — APPROVED** [review]

**Comment by kimikage** [comment]
Thanks!

**Comment by hyrodium** [comment]
Thank you for maintaining this repository!

**Comment by kimikage** [comment]
> The cause of the julia-invalidations error has not been identified, but is unrelated to this PR.

```
ERROR: Unsatisfiable requirements detected for package FlameGraphs [08572546]:
 FlameGraphs [08572546] log:
 ├─possible versions are: 0.1.0-1.0.0 or uninstalled
 ├─restricted by compatibility requirements with FixedPointNumbers [53c48c17] to versions: uninstalled
 │ └─FixedPointNumbers [53c48c17] log:
 │   ├─possible versions are: 0.9.0 or uninstalled
 │   └─FixedPointNumbers [53c48c17] is fixed to version 0.9.0-dev
```
🤣

**Pull request merged**